### PR TITLE
Add `random` value to block metadata and fix `sys_prevrandao`

### PR DESCRIPTION
**Pull request by wborgeaud**
See #1204 for details.

### CODE REVIEW

File: evm/src/proof.rs [hunk 1]
```diff
@@ -68,6 +68,7 @@ pub struct BlockMetadata {
     pub block_timestamp: U256,
     pub block_number: U256,
     pub block_difficulty: U256,
+    pub block_random: U256,
```
Comment (Nashtare) [inline evm/src/proof.rs:71]: Perhaps an overlook, but shouldn't this rather be an `H256` out of consistency with `mix_hash` output?

File: evm/src/cpu/kernel/asm/core/syscall_stubs.asm
Comment (LindaGuiga) [inline evm/src/cpu/kernel/asm/core/syscall_stubs.asm:1]: Should this be removed from `syscall_stubs.asm` and put in (for example) `metadata.asm` instead?

File: evm/src/proof.rs [hunk 2]
```diff
@@ -336,6 +346,9 @@ impl BlockMetadataTarget {
         builder.connect(bm0.block_timestamp, bm1.block_timestamp);
         builder.connect(bm0.block_number, bm1.block_number);
         builder.connect(bm0.block_difficulty, bm1.block_difficulty);
+        for i in 0..8 {
+            builder.connect(bm0.block_random[i], bm1.block_random[i]);
```
Comment (LindaGuiga) [inline evm/src/proof.rs:350]: `block_random` is connected here, but it isn't set in `set_block_metadata_target`. This leads to a `generators not run` error in a recursive proof.

File: evm/src/cpu/kernel/asm/core/syscall_stubs.asm [hunk 3]
```diff
@@ -10,11 +10,9 @@ global sys_blockhash:
     EXIT_KERNEL
 
 // This is a temporary version that returns the block difficulty (i.e. the old version of this opcode).
```
Comment (LindaGuiga) [inline evm/src/cpu/kernel/asm/core/syscall_stubs.asm:12]: Shouldn't this comment be deleted or updated?

**Review by LindaGuiga — COMMENTED** [review]
Thanks for implementing this! Looks great, but there's just one missing thing: `block_random` is not set in `set_block_metadata_target`.

**Review by Nashtare — APPROVED** [review]
Looks good!

**Pull request merged**